ksqldb-rest-app/src/main/java/io/confluent/ksql/api/impl/QueryEndpoint.java

```diff
@@ -183,9 +187,15 @@ private QueryPublisher createScalablePushQueryPublisher(
       final ServiceContext serviceContext,
       final ConfiguredStatement<Query> statement,
       final WorkerExecutor workerExecutor,
-      final Map<String, Object> requestProperties
+      final Map<String, Object> requestProperties,
+      final MetricsCallbackHolder metricsCallbackHolder
   ) {
-    final BlockingQueryPublisher publisher = new BlockingQueryPublisher(context, workerExecutor);
+    metricsCallbackHolder.setCallback((statusCode, requestBytes, responseBytes, startTimeNanos) -> {
+      scalablePushBandRateLimiter.add(responseBytes);
```

Comment:
I'm fairly certain this works as intended, but I want to be sure that these async endpoints only call this at the end, ensuring we get the total amount of bandwidth.  Do we have a test for the http2 endpoint?